### PR TITLE
【BIT】转换能力补齐：paddle.nn.functional.max_unpool[1 2 3]d

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. [paddle.nn.functional.max_unpool1d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/max_unpool1d_cn.html)

查询 [torch.nn.functional.max_unpool1d 与 paddle.nn.functional.max_unpool1d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.max_unpool1d.html) ，发现是 `paddle 参数更多`，因此编写 mapping 转换配置，注册 UnpoolRule 转换规则：

- paddle 多 data_format 参数，使用 permute 调整维度顺序
- paddle 与 torch 参数用法不一致，kernel_size、stride、padding、output_size 若为 list，需转为 tuple；indices 转为 int64 类型

### 2. [paddle.nn.functional.max_unpool2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/max_unpool2d_cn.html)

查询 [torch.nn.functional.max_unpool2d 与 paddle.nn.functional.max_unpool2d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.max_unpool2d.html) ，发现是 `paddle 参数更多`，因此编写 mapping 转换配置，注册 UnpoolRule 转换规则

### 3. [paddle.nn.functional.max_pool3d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/max_pool3d_cn.html)

查询 [torch.nn.functional.max_unpool3d 与 paddle.nn.functional.max_unpool3d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.max_unpool3d.html) ，发现是 `paddle 参数更多`，因此编写 mapping 转换配置，注册 UnpoolRule 转换规则

## 测试结果

在 https://github.com/PFCCLab/PaddleAPITest/pull/116 修复后，paddle.nn.functional.max_unpool[1 2 3]d 的所有配置均通过 accuracy 测试，移入 support2torch_1

image

### CODE REVIEW

**Review by Cutelemon6 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
